tfx/examples/imdb/imdb_pipeline_native_keras.py

```diff
@@ -0,0 +1,196 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""IMDB Sentiment Analysis example using TFX."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+from typing import Text
+
+import absl
+import tensorflow_model_analysis as tfma
+
+from tfx.components import CsvExampleGen
+from tfx.components import Evaluator
+from tfx.components import ExampleValidator
+from tfx.components import Pusher
+from tfx.components import ResolverNode
+from tfx.components import SchemaGen
+from tfx.components import StatisticsGen
+from tfx.components import Trainer
+from tfx.components import Transform
+from tfx.components.trainer.executor import GenericExecutor
+from tfx.components.base import executor_spec
+from tfx.utils.dsl_utils import external_input
+from tfx.dsl.experimental import latest_blessed_model_resolver
+from tfx.types import Channel
+from tfx.types.standard_artifacts import Model
+from tfx.types.standard_artifacts import ModelBlessing
+
+from tfx.proto import example_gen_pb2
+from tfx.proto import pusher_pb2
+from tfx.proto import trainer_pb2
+
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+
+_pipeline_name = 'imdb_native_keras'
+
+# This example assumes that IMDB review data is stored in ~/imdb/data and the
+# utility function is in ~/imdb. Feel free to customize as needed.
+_imdb_root = os.path.join(os.environ['HOME'], 'imdb')
+_data_root = os.path.join(_imdb_root, 'data')
+# Python module file to inject customized logic into the TFX components. The
+# Transform and Trainer both require user-defined functions to run successfully.
+_module_file = os.path.join(_imdb_root, 'imdb_utils_native_keras.py')
+# Path which can be listened to by the model server.  Pusher will output the
+# trained model here.
+_serving_model_dir = os.path.join(_imdb_root, 'serving_model', _pipeline_name)
+
+# Directory and data locations.  This example assumes all of the
+# example code and metadata library is relative to $HOME, but you can store
+# these files anywhere on your local filesystem.
+_tfx_root = os.path.join(os.environ['HOME'], 'tfx')
+_pipeline_root = os.path.join(_tfx_root, 'pipelines', _pipeline_name)
+# Sqlite ML-metadata db path.
+_metadata_path = os.path.join(_tfx_root, 'metadata', _pipeline_name,
+                              'metadata.db')
+
+# Pipeline arguments for Beam powered Components.
+_beam_pipeline_args = [
+    '--direct_running_mode=multi_processing',
+    # 0 means auto-detect based on on the number of CPUs available
+    # during execution time.
+    '--direct_num_workers=1',
+]
+
+def _create_pipeline(pipeline_name: Text, pipeline_root: Text, data_root: Text,
+                     module_file: Text, serving_model_dir: Text,
+                     metadata_path: Text) -> pipeline.Pipeline:
```

Comment:
could you pass in the _beam_pipeline_args like other examples?

Reply:
yes! I think i was looking at an old example the last time I updated this